src/eval/str.rs

```diff
@@ -521,3 +582,17 @@ cast_from_value! {
         _ => Err("expected either `start` or `end`")?,
     },
 }
+
+/// A replacement for a matched [`Str`]
+pub enum Replacement {
+    Str(Str),
+    /// Either of type `Dict -> Str` (see `captures_to_dict`) if the match occured
+    /// by regex or `Str -> Str` if matched by Str.
```

Comment:
I think the function should consistently operate on match dictionaries or string regardless of whether a string or regex is used as the search pattern.

Reply:
I would argue that we should pass it a match dictionary in any case then because otherwise one cannot utilise capture groups of regexes properly.